Sources/AsyncHTTPClient/NIOTransportServices/NWErrorHandler.swift

```diff
@@ -0,0 +1,172 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the AsyncHTTPClient open source project
+//
+// Copyright (c) 2020 Apple Inc. and the AsyncHTTPClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of AsyncHTTPClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if canImport(Network)
+
+    import Network
+    import NIO
+    import NIOHTTP1
+    import NIOTransportServices
+
+    /// Enum containing all the DNS errors
+    public enum NWDNSError: Error {
+        case noError
+        case unknown /* 0xFFFE FFFF */
+        case noSuchName
+        case noMemory
+        case badParam
+        case badReference
+        case badState
+        case badFlags
+        case unsupported
+        case notInitialized
+        case alreadyRegistered
+        case nameConflict
+        case invalid
+        case firewall
+        case incompatible /* client library incompatible with daemon */
+        case badInterfaceIndex
+        case refused
+        case noSuchRecord
+        case noAuth
+        case noSuchKey
+        case NATTraversal
+        case doubleNAT
+        case badTime /* Codes up to here existed in Tiger */
+        case badSig
+        case badKey
+        case transient
+        case serviceNotRunning /* Background daemon not running */
+        case NATPortMappingUnsupported /* NAT doesn't support PCP, NAT-PMP or UPnP */
+        case NATPortMappingDisabled /* NAT supports PCP, NAT-PMP or UPnP, but it's disabled by the administrator */
+        case noRouter /* No router currently configured (probably no network connectivity) */
+        case pollingMode
+        case timeout
+        case defunctConnection /* Connection to daemon returned a SO_ISDEFUNCT error result */
+        case other(DNSServiceErrorType)
```

Comment:
`enum`s with an `other` case usually should really be `struct`s because equality/hashability/extensibility is a problem for enums.

Reply:
this is no longer an issue as I deleted it